File: Python/6py 2/steps_in_primes.py
"""
|--------------------------------------------------------------------------
| Steps in Primes
|--------------------------------------------------------------------------
|
"""

def is_prime(n):
   for i in range(2,int(n**0.5) + 1):
      if n % i == 0: return False
   return True
def step(g, m, n):
   for i in range(m, n - g):
      if is_prime(i) and is_prime(i + g): return [i,i+g]
   return None
         

from gmpy2 import is_prime
def step(g, m, n):
   for i in range(m, n-g+1):
      if is_prime(i) and is_prime(i+g):
         return [i, i+g]
      

from gmpy2 import next_prime, is_prime
def step(g, m, n):
   m = next_prime(m-1)
   while m<n:
      if is_prime(g+m) and g+m<=n:
         return [m, g+m]
      m = next_prime(m)


print(step(2,100,110), [101, 103])
print(step(4,100,110), [103, 107])
print(step(2,5,5), None)
print(step(6,100,110), [101, 107])
print(step(8,300,400), [359, 367])
print(step(10,300,400), [307, 317])

File: PHP/6php 2/steps_in_primes.php
<?php
/*
|--------------------------------------------------------------------------
| Steps in Primes
|--------------------------------------------------------------------------
|
*/

function is_prime($n) {
   if ($n <= 1) return false;
   if ($n <= 3) return true;
   if ($n % 2 == 0 || $n % 3 == 0) return false;
   for ($i = 5; $i * $i <= $n; $i += 6) {
      if ($n % $i == 0 || $n % ($i + 2) == 0) return false;
   }
   return true;
}
function step($g, $m, $n) {
   for ($i = $m; $i <= $n - $g; $i++) {
      if (is_prime($i) && is_prime($i + $g)) {
         return [$i, $i + $g];
      }
   }
   return null;
}


function isPrime($n) {
   for ($i=2; $i*$i<=$n; $i++){
      if ($n%$i==0) return false;
   }
   return $n>=2;
}
function step($g, $m, $n) {
   while ($m+$g<=$n){
      if (isPrime($m) && isPrime($m+$g)) {
         return [$m, $m+$g];
      }
      $m++;
   }
   return null;
}


print_r(step(2, 100, 110));
print_r(step(4, 100, 110));
print_r(step(6, 100, 110));
print_r(step(8, 300, 400));
print_r(step(10, 300, 400));